tests/include/mock_nrf_modem_at.h

```diff
@@ -35,8 +35,36 @@ void mock_nrf_modem_at_Init(void);
  */
 void mock_nrf_modem_at_Verify(void);
 
+#if defined(CONFIG_MOCK_NRF_MODEM_AT_PRINTF)
+
+/**
+ * @brief Mock for nrf_modem_at_scanf().
+ *
+ * This function will verify that the string produced from @ref fmt using vsprintf()
+ * with variable arguments matches the expected string.
+ *
+ * @param[in] fmt Format.
+ */
+int __cmock_nrf_modem_at_printf(const char *fmt, ...);
+
+/**
+ * @brief Sets expected parameter and return value.
+ *
+ * @param[in] fmt Expected string. This is not the format with specifiers but the
+ *                expected string after applying variable arguments with vsprintf().
+ * @param[in] return_value Expected return value.
```

Comment:
This is the value the mock returns, I think the word "expected" should be removed. The same applies to `__mock_nrf_modem_at_scanf_ExpectAndReturn()`.